File: src/com/dafttech/terra/graphics/passes/RenderingPass.java
package com.dafttech.terra.graphics.passes;

import com.dafttech.terra.game.world.World;
import com.dafttech.terra.game.world.entities.Entity;
import com.dafttech.terra.graphics.AbstractScreen;

public abstract class RenderingPass {
    public static PassObjects rpObjects = new PassObjects();
    public static PassLighting rpLighting = new PassLighting();
    public static PassGaussian rpGaussian = new PassGaussian();
    public static PassGUIContainer rpGUIContainer = new PassGUIContainer();

    public abstract void applyPass(AbstractScreen screen, Entity pointOfView, World w, Object... arguments);
}

File: terra/src/main/scala/com/dafttech/terra/engine/passes/RenderingPass.scala
package com.dafttech.terra.engine.passes

import com.dafttech.terra.engine.AbstractScreen
import com.dafttech.terra.game.Events
import com.dafttech.terra.game.world.GameWorld
import com.dafttech.terra.game.world.entities.Entity

object RenderingPass {
  val rpObjects = new PassObjects
  val rpLighting = new PassLighting
  val rpGaussian = new PassGaussian
  val rpGUIContainer = new PassGUIContainer
  val rpBackground = new PassBackgrounds
}

abstract class RenderingPass {
  def applyPass(screen: AbstractScreen, pointOfView: Entity, w: GameWorld, arguments: AnyRef*)

  Events.EVENTMANAGER.registerEventListener(this)
}